cmd/env.go

```diff
@@ -1,61 +1,48 @@
 package cmd
 
 import (
-	"context"
-	"fmt"
-
 	"github.com/celestiaorg/celestia-node/node"
+
+	"context"
 )
 
-// Env is an environment for CLI commands.
-// It can be used to:
-// 1. Propagate values from parent to child commands.
-// 2. To group common logic that multiple commands rely on.
-// Usage can be extended.
-// TODO(@Wondertan): We should move to using context only instead.
-//  Env, in fact, only keeps some additional fields which should be
-//  kept in the context directly using WithValue (#965)
-type Env struct {
-	NodeType  node.Type
-	StorePath string
-
-	opts []node.Option
+// NodeType reads the node.Type from the context.
+func NodeType(ctx context.Context) node.Type {
+	return ctx.Value(nodeTypeKey{}).(node.Type)
 }
 
-// WithEnv wraps given ctx with Env.
-func WithEnv(ctx context.Context) context.Context {
-	_, err := GetEnv(ctx)
-	if err == nil {
-		panic("cmd: only one Env is allowed to be set in a ctx")
-	}
-
-	return context.WithValue(ctx, envCtxKey{}, &Env{})
+// StorePath reads the store path from the context.
+func StorePath(ctx context.Context) string {
+	return ctx.Value(storePathKey{}).(string)
 }
 
-// GetEnv takes Env from the given ctx, if any.
-func GetEnv(ctx context.Context) (*Env, error) {
-	env, ok := ctx.Value(envCtxKey{}).(*Env)
-	if !ok {
-		return nil, fmt.Errorf("cmd: Env is not set in ctx.Context")
-	}
-
-	return env, nil
+// SetNodeType sets Node Type to the Env.
+func SetNodeType(ctx context.Context, tp node.Type) context.Context {
+	return context.WithValue(ctx, nodeTypeKey{}, tp)
 }
 
-// SetNodeType sets Node Type to the Env.
-func (env *Env) SetNodeType(tp node.Type) {
-	env.NodeType = tp
+// SetStorePath sets Store Path to the Env.
+func SetStorePath(ctx context.Context, storePath string) context.Context {
+	return context.WithValue(ctx, storePathKey{}, storePath)
 }
 
 // Options returns Node Options parsed from Environment(Flags, ENV vars, etc)
-func (env *Env) Options() []node.Option {
-	return env.opts
+func Options(ctx context.Context) []node.Option {
```

Comment:
NodeOptions?

Reply:
Renamed to NodeOptions, but one question: Should AddNodeOptions also be refactored to WithNodeOptions? I don't want it to be unclear that the previous ones are not being overwritten, but just appended